src/ol/renderer/Composite.js

```diff
@@ -110,15 +110,17 @@ class CompositeMapRenderer extends MapRenderer {
     for (let i = 0, ii = layerStatesArray.length; i < ii; ++i) {
       const layerState = layerStatesArray[i];
       frameState.layerIndex = i;
+
+      const layer = layerState.layer;
+      const sourceState = layer.getSourceState();
       if (
         !inView(layerState, viewState) ||
-        (layerState.sourceState != SourceState.READY &&
-          layerState.sourceState != SourceState.UNDEFINED)
+        (sourceState != SourceState.READY &&
+          sourceState != SourceState.UNDEFINED)
```

Comment:
This change allows the removal of the `sourceState` property of the layer state. There were no other call sites that used this property. Now the layer's `getSourceState()` method is used everywhere.